raster-foundry/raster-foundry
# Add isActive to AOIs

## Overview

This PR adds `is_active` to AOIs in an attempt to mitigate the upcoming AOIpocalypse.
AOIs haven't been running successfully for quite a long time, meanwhile, we've imported
literally millions of scenes. Once they're running successfully, that will almost inevitably
and unstoppably crush the ingest queue. That's bad! By adding an `is_active` parameter,
defaulting new AOIs to active, and filtering out projects with inactive AOIs in the
`FindAOIProject`s step, it'll be a lot like we have the time stone and can pause everything
until we're ready.

### Checklist

- [x] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated (raster-foundry/raster-foundry-api-spec#10)

## Testing Instructions

 * Observe that AOI tests still pass in CI
 * Reassemble your batch jar
 * Run migrations
 * Make yourself a nice AOI project in the frontend
 * Run `java -cp /opt/raster-foundry/jars/rf-batch.jar com.azavea.rf.batch.Main find_aoi_projects` in the batch container: it should only find the project you just created
 * `update aois set is_active = false;`
 * Run the `find_aoi_projects` command again
 * It shouldn't find anything

Closes azavea/raster-foundry-platform#305
